truffle/src/main/ruby/core/rubinius/api/shims/mirror.rb

```diff
@@ -31,6 +30,7 @@ def self.module_mirror(obj)
         end
       end
     end
+    Truffle::Primitive.install_rubinius_primitive self, :module_mirror
```

Comment:
Yes I thought that `self` was the module when it isn't. I think I will use some magic to get the self of the caller, since this slow path stuff.